### PR TITLE
Porting Tarred ASR Dataset

### PR DESCRIPTION
**Pull request by redoctopus**
Adds Tarred ASR Dataset (TarredAudioToCharDataset) support to NeMo v1.0.

This has mostly equivalent behavior to the TarredAudioToTextDataLayer from previous versions, taking a manifest path and one or more tarred audio dataset paths from which to load samples. Documentation has been updated and clarified, and a few extra fields added to the example QuartzNet config.

Note that `limit_train_batches` needs to be set (as in `ctc_models.py`) for tqdm to display training progress properly when using this Dataset, since the length of the DataLoader with an IterableDataset is the length of the Dataset (if given) rather than the number of batches.

### CODE REVIEW

**Comment by okuchaiev** [comment]
there is also a conflict with base branch which needs resolving

File: nemo/collections/asr/data/audio_to_text.py [hunk 1]
```diff
@@ -11,17 +11,22 @@
...
 # limitations under the License.
+import io
+import os
 from typing import Callable, Dict, List, Optional, Union
 
+import braceexpand
 import torch
+import webdataset as wd
 
+from nemo import logging
```
Comment (blisc) [inline nemo/collections/asr/data/audio_to_text.py:22]: It is now `from nemo.utils import logging`

File: nemo/collections/asr/data/audio_to_text.py [hunk 2]
```diff
@@ -205,8 +203,10 @@ class AudioToCharDataset(_AudioTextDataset):
         manifest_filepath: Path to manifest json as described above. Can
             be comma-separated paths.
         labels: String containing all the possible characters to map to
-        featurizer: Initialized featurizer class that converts paths of
-            audio to feature tensors
+        sample_rate (int): Sample rate to resample loaded audio to
+        int_values (bool): If true, load samples as 32-bit integers. Defauts to False.
+        augmentor (nemo.collections.asr.parts.perturb.AudioAugmentor): An AudioAugmentor
+            object used to augment loaded audio
```
Comment (blisc) [inline nemo/collections/asr/data/audio_to_text.py:209]: Thanks for filling this out

File: nemo/collections/asr/models/ctc_models.py [hunk 3]
```diff
@@ -47,6 +48,13 @@ def list_available_models(cls) -> Optional[PretrainedModelInfo]:
         return result
 
     def __init__(self, cfg: DictConfig, trainer: Trainer = None):
+        # Get global rank and total number of GPU workers for IterableDataset partitioning, if applicable
+        self.global_rank = 0
+        self.world_size = 0
+        if trainer is not None:
+            self.global_rank = (trainer.node_rank * trainer.num_processes) + trainer.local_rank
+            self.world_size = trainer.num_nodes * trainer.num_processes
```
Comment (blisc) [inline nemo/collections/asr/models/ctc_models.py:56]: Let's use trainer.num_gpus instead

Comment (redoctopus) [reply nemo/collections/asr/models/ctc_models.py:56]: Just wondering--is there a significant difference here? The PTL trainer documentation that I've found hasn't been terribly illuminating, and only gives some details about `num_processes` while having very little about `num_gpus`.

File: nemo/collections/asr/data/audio_to_text.py (hunk 1 above)
Comment (redoctopus) [reply nemo/collections/asr/data/audio_to_text.py]: Updated

**Comment by blisc** [comment]
Please merge when ready

**Pull request merged**